File: java-example/src/test/java/org/onflow/examples/java/verifySignature/userSignatureValidateAll/UserSignatureValidateAllExampleTest.java
package org.onflow.examples.java.verifySignature.userSignatureValidateAll;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.onflow.flow.common.test.FlowEmulatorProjectTest;
import org.onflow.flow.common.test.FlowTestAccount;
import org.onflow.flow.common.test.FlowTestClient;
import org.onflow.flow.common.test.TestAccount;
import org.onflow.flow.sdk.FlowAccessApi;
import org.onflow.flow.sdk.cadence.BooleanField;
import org.onflow.flow.sdk.cadence.Field;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.fail;

@FlowEmulatorProjectTest(flowJsonLocation = "../flow/flow.json")
public class UserSignatureValidateAllExampleTest {
    @FlowTestAccount
    private TestAccount testAccount;
    @FlowTestClient
    private FlowAccessApi accessAPI;
    private UserSignatureValidateAllExample connector;

    @BeforeEach
    public void setup() {
        connector = new UserSignatureValidateAllExample(accessAPI);
    }

    @Test
    public void canVerifyUserSignature() {
        try {
            Field<?> txResult = connector.verifyUserSignatureValidateAll(testAccount.getFlowAddress(), testAccount.getPrivateKey());

            if (txResult instanceof BooleanField) {
                assertEquals(Boolean.TRUE, ((BooleanField) txResult).getValue(), "Signature verification failed");
            } else {
                fail("Expected BooleanField but got " + txResult.getClass().getSimpleName());
            }
        } catch (Exception e) {
            fail("Test failed with exception: " + e.getMessage());
        }
    }
}

File: kotlin-example/src/test/kotlin/org/onflow/examples/kotlin/verifySiganture/userSignatureValidateAll/UserSignatureValidateAllExampleTest.kt
package org.onflow.examples.kotlin.verifySiganture.userSignatureValidateAll

import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Assertions.fail
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.onflow.examples.kotlin.verifySignature.userSignatureValidateAll.UserSignatureValidateAllExample
import org.onflow.flow.common.test.*
import org.onflow.flow.sdk.*
import org.onflow.flow.sdk.cadence.BooleanField

@FlowEmulatorProjectTest(flowJsonLocation = "../flow/flow.json")
internal class UserSignatureValidateAllExampleTest {
    @FlowTestAccount
    lateinit var testAccount: TestAccount

    @FlowTestClient
    lateinit var accessAPI: FlowAccessApi

    private lateinit var connector: UserSignatureValidateAllExample

    @BeforeEach
    fun setup() {
        connector = UserSignatureValidateAllExample(accessAPI)
    }

    @Test
    fun `Can verify user signature`() {
        val txResult = connector.verifyUserSignatureValidateAll(testAccount.flowAddress, testAccount.privateKey)

        if (txResult is BooleanField) {
            assertTrue(txResult.value!!, "Signature verification failed")
        } else {
            fail("Expected BooleanField but got ${txResult::class.simpleName}")
        }
    }
}
